src/action.cpp

```diff
@@ -599,8 +599,8 @@ bool can_butcher_at( const tripoint &p )
 {
     Character &player_character = get_player_character();
     // TODO: unify this with game::butcher
-    const int factor = player_character.max_quality( qual_BUTCHER );
-    const int factorD = player_character.max_quality( qual_CUT_FINE );
+    const int factor = player_character.max_quality( qual_BUTCHER, PICKUP_RANGE );
```

Comment:
```suggestion
    const int factor = player_character.crafting_inventory( true ).max_quality( qual_BUTCHER );
```

It should be something like this.

Reply:
`crafting_inventory` iterate over items, `Character::max_quality` also checks the bionics and mutations for the quality.

Reply:
If `crafting_inventory` is missing any of the bionics or mutations, that's a bug. It's supposed to include anything the character has, too.

Reply:
In that case you should open a new PR to fix it, as it would be out of scope for this PR.

Reply:
I'm not one to decide that, but you could argue it's within the scope of this PR, if this prevents you from making a proper solution. What you're trying to solve _should_ be part of what the `inventory` functionality provides.

At the very least I would say move your new method out of visitable.cpp, because it really does not belong there. It's not an expansion of the `visitable` infrastructure, it's just a `Character` method that uses it.

Reply:
Saicchi: Perhaps inquire on Discord re new PR or not? Up to you, of course!